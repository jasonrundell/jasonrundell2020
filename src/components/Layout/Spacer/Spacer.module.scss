@import '../../../../src/styles/main.scss';

$sizes: (
  smallest: $sizeSmallest,
  smaller: $sizeSmaller,
  small: $sizeSmall,
  normal: $sizeNormal,
  large: $sizeLarge,
  larger: $sizeLarger,
  largest: $sizeLargest,
);

@each $name, $size in $sizes {
  .small-#{$name} {
    height: $size;
  }
  .medium-#{$name} {
    @media (min-width: $mediumBreakpoint) {
      height: $size;
    }
  }
  .large-#{$name} {
    @media (min-width: $largeBreakpoint) {
      height: $size;
    }
  }
}

.root {
}
