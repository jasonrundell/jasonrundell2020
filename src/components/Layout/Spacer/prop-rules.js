const heightRules = [
  'smallest',
  'smaller',
  'small',
  'normal',
  'large',
  'larger',
  'largest',
]

export default { heightRules }
