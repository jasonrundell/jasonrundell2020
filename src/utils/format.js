/**
 *  Letter case conversion functions
 *  [Letter case](https://en.wikipedia.org/wiki/Letter_case#Special_case_styles)
 *
 *  Converts only letter case-separated or hyphen-delimited strings
 */

/**
 *  Converts hyphen-delimited `string` to camel case
 *
 *  @example
 *  'foo-bar' => 'fooBar'
 *  'foo--bar' => 'fooBar'
 *
 *  @param {string} string
 */
function toCamelCase(string) {
  return string.replace(/(-+)([a-z])/g, ($0, $1, $2) => $2.toUpperCase())
}

/**
 *  Converts hyphen-delimited `string` to pascal case (aka upper camel case)
 *
 *  @example
 *  'foo-bar' => 'FooBar'
 *  'foo--bar' => 'FooBar'
 *
 *  @param {string} string
 */
function toPascalCase(string) {
  return toCamelCase(string.replace(/^[a-z]/, $0 => $0.toUpperCase()))
}

export { toCamelCase, toPascalCase }
