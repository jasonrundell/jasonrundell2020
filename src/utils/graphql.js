export function edgesToObject(edgesData) {
  const nodes = edgesData.map(item => {
    return item.node
  })
  const finalObject = nodes.map(item => {
    return item
  })

  return finalObject
}
