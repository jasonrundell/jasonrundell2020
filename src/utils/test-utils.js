export function catchWarnings() {
  const originalConsoleError = global.console.error
  beforeEach(() => {
    /**
     * I Want to know if values passed to props fail the type check
     * */
    global.console.error = (...args) => {
      const propTypeFailures = [/Warning:/] // e.g. [/Invalid prop/, /Failed prop type/]

      if (propTypeFailures.some(p => p.test(args[0]))) {
        throw new Error(args[0])
      }

      originalConsoleError(...args)
    }
  })
}

/**
 * Mix an match help from:
 * https://daveceddia.com/testing-for-proptypes-errors/
 * https://www.jackfranklin.co.uk/blog/failing-tests-on-react-proptypes/
 */
