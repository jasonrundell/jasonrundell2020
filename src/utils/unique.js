const onlyUnique = (value, index, self) => {
  return self.indexOf(value) === index
}

const generatePassword = ({
  passwordLength = 16,
  charset = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789!@#$%^&*()_-=~`',
}) => {
  let retVal = ''
  for (var i = 0, n = charset.length; i < passwordLength; ++i) {
    retVal += charset.charAt(Math.floor(Math.random() * n))
  }
  return retVal
}

export { onlyUnique, generatePassword }
