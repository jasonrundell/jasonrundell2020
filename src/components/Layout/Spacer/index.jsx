import React from 'react'
import PropTypes from 'prop-types'

import { heightRules } from './prop-rules'

import { classNames } from '../../../utils/css'
import { toCamelCase } from '../../../utils/format'

import {
  root as rootStyle,
  largeLarge,
  largeLarger,
  largeLargest,
  largeNormal,
  largeSmall,
  largeSmaller,
  largeSmallest,
  mediumLarge,
  mediumLarger,
  mediumLargest,
  mediumNormal,
  mediumSmall,
  mediumSmaller,
  mediumSmallest,
  smallLarge,
  smallLarger,
  smallLargest,
  smallNormal,
  smallSmall,
  smallSmaller,
  smallSmallest,
} from './Spacer.module.scss'

const Spacer = ({ sizeSmall, sizeMedium, sizeLarge }) => {
  const classes = classNames(
    rootStyle,
    sizeSmall === 'smallest' && smallSmallest,
    sizeSmall === 'smaller' && smallSmaller,
    sizeSmall === 'small' && smallSmall,
    sizeSmall === 'normal' && smallNormal,
    sizeSmall === 'large' && smallLarge,
    sizeSmall === 'larger' && smallLarger,
    sizeSmall === 'largest' && smallLargest,
    sizeMedium === 'smallest' && mediumSmallest,
    sizeMedium === 'smaller' && mediumSmaller,
    sizeMedium === 'small' && mediumSmall,
    sizeMedium === 'normal' && mediumNormal,
    sizeMedium === 'large' && mediumLarge,
    sizeMedium === 'larger' && mediumLarger,
    sizeMedium === 'largest' && mediumLargest,
    sizeLarge === 'smallest' && largeSmallest,
    sizeLarge === 'smaller' && largeSmaller,
    sizeLarge === 'small' && largeSmall,
    sizeLarge === 'normal' && largeNormal,
    sizeLarge === 'large' && largeLarge,
    sizeLarge === 'larger' && largeLarger,
    sizeLarge === 'largest' && largeLargest
  )
  return <div className={classes} />
}

Spacer.defaultProps = {
  sizeSmall: heightRules[2],
  sizeMedium: heightRules[3],
  sizeLarge: heightRules[4],
}

Spacer.propTypes = {
  sizeSmall: PropTypes.oneOf(heightRules),
  sizeMedium: PropTypes.oneOf(heightRules),
  sizeLarge: PropTypes.oneOf(heightRules),
}

export default Spacer
