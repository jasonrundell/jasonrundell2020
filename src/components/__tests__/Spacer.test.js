import React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { act } from 'react-dom/test-utils'
import renderer from 'react-test-renderer'
import { catchWarnings } from '../../utils/test-utils'

import Spacer from '../Layout/Spacer'
import Rules from '../Layout/Spacer/prop-rules'

const { heightRules } = Rules

catchWarnings()

let container = null
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  unmountComponentAtNode(container)
  container.remove()
  container = null
})

describe('Spacer component', () => {
  it('Spacer renders correctly', () => {
    const tree = renderer
      .create(
        <div>
          <Spacer />
          <Spacer sizeLarge={heightRules[0]} />
          <Spacer sizeLarge={heightRules[heightRules.length - 1]} />
        </div>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
