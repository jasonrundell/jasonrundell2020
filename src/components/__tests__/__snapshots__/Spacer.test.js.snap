// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer component Spacer renders correctly 1`] = `
<div>
  <div
    className="root smallSmall mediumNormal largeLarge"
  />
  <div
    className="root smallSmall mediumNormal largeSmallest"
  />
  <div
    className="root smallSmall mediumNormal largeLargest"
  />
</div>
`;
